src/lightning_app/utilities/commands/base.py

```diff
@@ -0,0 +1,170 @@
+    command_metadata: Dict[str, Any], app_id: Optional[str]
+) -> Tuple[ClientCommand, Dict[str, BaseModel]]:
+    config = _Config(**command_metadata)
+    tmpdir = osp.join(gettempdir(), f"{getuser()}_commands")
+    makedirs(tmpdir)
+    target_file = osp.join(tmpdir, f"{config.command}.py")
+    if app_id:
+        client = LightningClient()
+        project_id = _get_project(client).project_id
+        response = client.lightningapp_instance_service_list_lightningapp_instance_artifacts(project_id, app_id)
+        for artifact in response.artifacts:
+            if f"commands/{config.command}.py" == artifact.filename:
+                r = requests.get(artifact.url, allow_redirects=True)
+                with open(target_file, "wb") as f:
+                    f.write(r.content)
+    else:
+        shutil.copy(config.cls_path, target_file)
+
+    cls_name = config.cls_name
+    spec = spec_from_file_location(config.cls_name, target_file)
+    mod = module_from_spec(spec)
+    sys.modules[cls_name] = mod
+    spec.loader.exec_module(mod)
+    command = getattr(mod, cls_name)(method=None, requirements=config.requirements)
+    models = {k: getattr(mod, v) for k, v in config.params.items()}
+    shutil.rmtree(tmpdir)
+    return command, models
+
+
+def _to_annotation(anno: str) -> str:
+    anno = anno.split("'")[1]
+    if "." in anno:
+        return anno.split(".")[-1]
+    return anno
+
+
+def _command_to_method_and_metadata(command: ClientCommand) -> Tuple[Callable, Dict[str, Any]]:
+    """Extract method and its metadata from a ClientCommand."""
+    params = inspect.signature(command.method).parameters
+    command_metadata = {
+        "cls_path": inspect.getfile(command.__class__),
+        "cls_name": command.__class__.__name__,
+        "params": {p.name: _to_annotation(str(p.annotation)) for p in params.values()},
+        **command._to_dict(),
+    }
+    method = command.method
+    command.models = {}
+    for k, v in command_metadata["params"].items():
+        if v == "_empty":
+            raise Exception(
+                f"Please, annotate your method {method} with pydantic BaseModel. Refer to the documentation."
+            )
+        config = getattr(sys.modules[command.__module__], v, None)
+        if config is None:
+            config = getattr(sys.modules[method.__module__], v, None)
+            if config:
+                raise Exception(
+                    f"The provided annotation for the argument {k} should in the file "
+                    f"{inspect.getfile(command.__class__)}, not {inspect.getfile(command.method)}."
+                )
+        if config is None or not issubclass(config, BaseModel):
+            raise Exception(
+                f"The provided annotation for the argument {k} shouldn't an instance of pydantic BaseModel."
+            )
+        command.models[k] = config
+    return method, command_metadata
+
+
+def _upload_command(command_name: str, command: ClientCommand) -> Optional[str]:
```

Comment:
I feel like this is a reinventing of code distribution and pip packages but it's fine, it's a  🛹 

The way I see it in prod-grade will be:
1- Some commands require installing a client pip package
2- install the client package
3- run client command from the installed package (without any imports, just execute a CLI command from the client_package)
4- inside the client command, we call the multiple remote commands as simple as an API request

Reply:
Yes, 100 %. This is meant to get things done very quickly before we standardize this and unblock the HPO Sweep App.

Reply:
And ideally, we handle the pip packaging, so users don't have to even think about it.